src/main/java/com/ravlinko/concordion/extension/mockserver/listener/MockServerParsingListener.java

```diff
@@ -23,39 +23,28 @@
 
 	@Override
 	public void beforeParsing(Document document) {
-		visit(document.getRootElement());
+		visitBeforeParsing(document.getRootElement());
 	}
 
-	private void visit(Element elem) {
+	private void visitBeforeParsing(Element elem) {
 		Elements children = elem.getChildElements();
 
 		for (int i = 0; i < children.size(); i++) {
-			visit(children.get(i));
+			visitBeforeParsing(children.get(i));
 		}
 
 		if (mockserverExtensionNs.equals(elem.getNamespaceURI())) {
-			String localName = translateTag(elem.getLocalName());
-			if (localName.isEmpty()) {
-				elem.getParent().removeChild(elem);
-			} else {
-				Attribute attr = new Attribute(elem.getLocalName(), "");
-				attr.setNamespace("ms", mockserverExtensionNs);
-				elem.addAttribute(attr);
-				elem.setNamespacePrefix("");
-				elem.setNamespaceURI(null);
-				elem.setLocalName(localName);
-			}
+			beforeParsingTag(elem.getLocalName()).ifPresent(mst -> mst.beforeParsing(elem));
 		}
 	}
 
-	private String translateTag(final String localName) {
+	private Optional<MockServerTag> beforeParsingTag(final String localName) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "beforeParsingTag" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)